src/PeopleSDKAdapter.js

```diff
@@ -0,0 +1,119 @@
+  }
+
+  /**
+   * Returns a proper Enum key for a given PersonStatus value
+   *
+   * @param {string} status
+   * @returns {string} PersonStatus
+   * @memberof PeopleSDKAdapter
+   */
+  getStatus(status) {
+    return Object.keys(PersonStatus).find((key) => PersonStatus[key] === status);
+  }
+
+  /**
+   * Encodes the personID into base64 format and return the UUID
+   *
+   * @param {string} ID
+   * @returns {string}
+   * @memberof PeopleSDKAdapter
+   */
+  getPersonUUID(ID) {
+    return Buffer.from(ID, 'base64')
+      .toString()
+      .split('/')[4]; // ciscospark://us/PEOPLE/<UUID>
+  }
+
+  /**
+   * Fetches the person data from the sdk and returns in the shape required by adapter.
+   *
+   * @param {string} ID
+   * @returns {Person}
+   * @memberof PeopleSDKAdapter
+   */
+  async fetchPerson(ID) {
+    const {id, emails, displayName, firstName, lastName, avatar, orgId} = await this.datasource.people.get(ID);
+
+    return {
+      ID: id,
+      emails,
+      displayName,
+      firstName,
+      lastName,
+      avatar,
+      orgID: orgId,
+    };
+  }
+
+  /**
+   * Returns an observable that emits person data of the given ID.
+   *
+   * @param {string} ID ID of person to get.
+   * @returns {Observable.<Person>}
+   * @memberof PeopleSDKAdapter
+   */
+  getPerson(ID) {
+    if (!(ID in this.getPersonObservables)) {
+      const personObs = from(this.fetchPerson(ID)).pipe(
+        map((data) => data),
+        (error) => error
+      );
+      const presenceObs = from(this.datasource.internal.presence.subscribe(this.getPersonUUID(ID))).pipe(
+        map((data) => data.responses[0].status.status, (error) => error)
+      );
+
+      // Subscribe to our internal `Apheleia` server to listen for any status change
+      // And update the person object with status response from the subscription
+      const personStatusObs = presenceObs.pipe(
+        flatMap((status) => personObs.pipe(map((person) => ({...person, status: this.getStatus(status)}))))
+      );
+
+      // Listen to any status change for the given person ID
+      const eventObs = fromEvent(this.datasource.internal.mercury, USER_PRESENCE_UPDATE_EVENT).pipe(
+        filter((event) => event.data.subject === this.getPersonUUID(ID)),
+        map((event) => this.getStatus(event.data.status))
+      );
+
+      // Update the person status after each change emitted from the event
+      const UpdatedPersonObs = personObs.pipe(
+        flatMap((person) => eventObs.pipe(map((status) => ({...person, status}))))
+      );
+
+      // Teardown logic
+      const teardownObs = Observable.create(() => async () => {
+        await this.datasource.internal.presence.unsubscribe(this.getPersonUUID(ID));
+      });
+
+      // fetch the original person data on the first run
+      // and updated object after each status change
+      // include the teardown logic to our finalized observable
+      const source = merge(personStatusObs, UpdatedPersonObs, teardownObs);
```

Comment:
`merge` is not the right operator here. If you look at the [marble diagram](http://reactivex.io/documentation/operators/merge.html), this operator combines the output of the observables as they come in, and not necessarily in the order you place them but that's not quite what as we cannot guarantee when an observable will complete.

We want to use [`concat`](https://rxjs-dev.firebaseapp.com/api/index/function/concat) instead, which guarantees order by waiting for the first observable to complete before going to the next.

So the `source` observable should look like:
```
const source = concat(personObs, statusUpdateObs);
```

Where the first `personObs` is created using [`fromPromise`](http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#static-method-fromPromise) specifically so it completes.

Also, I don't think the `teardown` logic is something we want in an observable, but maybe in an operator like [`finalize`](http://reactivex.io/rxjs/class/es6/Observable.js~Observable.html#instance-method-finalize)